autofit/graphical/expectation_propagation/optimiser.py

```diff
@@ -278,3 +354,118 @@ def _output_results(self, model_approx: EPMeanField):
         """
         with open(self.output_path / "graph.results", "w+") as f:
             f.write(self.factor_graph.make_results_text(model_approx))
+
+
+class StochasticEPOptimiser(EPOptimiser):
+
+    def factor_step(
+            self, 
+            factor, 
+            subset_approx, 
+            optimiser=None
+    ):
+        factor_logger = self.factor_loggers.get(factor.name, logging.getLogger(factor.name))
+        factor_logger.debug("Optimising...")
+
+        optimiser = optimiser or self.factor_optimisers[factor]
+        subset_factor = subset_approx._factor_subset_factor[factor]
```

Comment:
So the subset of factors remains the same each factor step?

Reply:
I'm not sure what you mean by this question. The subset of factors changes every batch.